File: backend/demo/src/main/java/com/example/models/Board.java
package com.example.models;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;

import java.time.LocalDateTime;
import java.util.ArrayList;

@Entity
public class Board {
    @Id
    @GeneratedValue
    private Long id;

    private Long ownerId;

    private LocalDateTime createdAt;

    private String name;
    private ArrayList<String> categories;

    /* will contain user id's */
    private ArrayList<Long> members;

}

File: frontend/src/Boards/Board.ts

import type { post } from "../Posts/Post"
// for reading a board
export interface Board {
    id: number;
    ownerId: string;
    name : string;
    description : string;
    categories: string[];
    members: string[];
    createdAt: Date;
    posts: post[];
}

// for creating a board
export interface createBoardRequest {
    name : string;
    description : string;
    categories: string[];
    members : string[];
}